src/dsql/parse.y

```diff
@@ -6833,6 +6836,34 @@ table_value_function_correlation_name
 	: as_noise symbol_item_alias_name	{ $$ = $2; }
 	;
 
+%type <recSourceNode> table_value_function_gen_series
+table_value_function_gen_series
+	: GENERATE_SERIES '(' table_value_function_gen_series_arg_list ')'
+		{
+			auto node = newNode<GenSeriesFunctionSourceNode>();
+			node->dsqlFlags |= RecordSourceNode::DFLAG_VALUE;
+			node->dsqlName = *$1;
+			node->inputList = $3;
+			node->dsqlField = nullptr;
+			$$ = node;
+		}
+	;
+
+%type <valueListNode> table_value_function_gen_series_arg_list
+table_value_function_gen_series_arg_list
+	: value ',' value step_opt
+		{
+			$$ = newNode<ValueListNode>($1);
+			$$->add($3);
+			$$->add($4);
+		}
+	;
+
+%type <valueExprNode> step_opt
```

Comment:
I'd call it `comma_value_opt`.

Reply:
No, default is fine if you pass it as a parameter. I.e. `comma_value_opt(1)` and define it as
```
comma_value_opt($default_value)
	: /* nothing */		{ $$ = MAKE_const_sint64($default_value, 0); }
	| ',' value			{ $$ = $2; }
	;

```

Reply:
Though name "optional_parameter" would be better.

Reply:
Well, you replaced the optional parameter with a predefined default value with an optional parameter with different values, but of a strictly defined type ((

Reply:
I'm not an expert in YACC so perhaps someone else can suggest how to pass node type and value via parameters to clause. May be something like `optional_parameter(LiteralNode, 1)` with `$$ = newNode<$node_type>($value)` inside...

Reply:
done